src/spaced/spaced/game/damage.hpp

```diff
@@ -1,17 +1,27 @@
 #pragma once
-#include <bave/core/time.hpp>
-#include <glm/vec2.hpp>
+#include <bave/imgui/im_text.hpp>
+#include <bave/app.hpp>
 
 namespace spaced {
 class Damage {
   public:
-	void inflict(float amount);
-	bool is_dead() const;
+	constexpr void inflict(int amount) { m_health -= glm::clamp(amount, amount, m_health); }
```

Comment:
It's better to clamp the result, especially once hit points are changed to be `float`s.
```cpp
m_health = std::clamp(m_health - amount, 0.0f, m_health /*or max_health*/);
```